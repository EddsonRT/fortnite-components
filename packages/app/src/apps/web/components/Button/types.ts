import { MouseEvent as ReactMouseEvent } from 'react';

export interface ButtonProps {
    label?: string;
    size?: 'small' | 'medium' | 'large';
    fullWidth?: boolean;
    style?: 'normal' | 'yellow';
    onClick?: (event: ReactMouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
