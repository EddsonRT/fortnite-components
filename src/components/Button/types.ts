export interface ButtonProps {
  label?: string
  fullWidth?: boolean
  size?: 'small' | 'medium' | 'large'
  style?:
    | 'lightblue'
    | 'blue'
    | 'yellow'
    | 'orange'
    | 'white'
    | 'black'
    | 'custom'
  textColor?: ''
  bgColor?: ''
  borderColor?: ''
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
}
