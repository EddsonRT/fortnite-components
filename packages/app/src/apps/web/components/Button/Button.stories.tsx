import React from 'react';
import { Story, Meta } from '@storybook/react';

import Button from './Button';
import { ButtonProps } from './types';

export default {
    title: 'Components/Button',
    component: Button,
    argTypes: {
        onClick: {
            action: 'clicked',
            table: {
                category: 'Events'
            }
        }
    },
    decorators: [
        (Story) => (
            <div tw='max-width[300px]'>
                <Story />
            </div>
        )
    ]
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

const label = 'Button';

export const Small = Template.bind({});
Small.args = {
    label,
    size: 'small'
};

export const Medium = Template.bind({});
Medium.args = {
    label
};

export const Large = Template.bind({});
Large.args = {
    label,
    size: 'large'
};

export const Yellow = Template.bind({});
Yellow.args = {
    label,
    style: 'yellow'
};
