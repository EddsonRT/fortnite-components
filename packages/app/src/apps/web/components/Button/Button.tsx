import React, { FC } from 'react';
import tw from 'twin.macro';

import { ButtonProps } from './types';

const Button: FC<ButtonProps> = ({
    label = '',
    fullWidth = false,
    size = 'medium',
    style = 'normal',
    onClick
}) => {
    return (
        <button
            className='group'
            css={[
                tw`cursor-pointer overflow-hidden focus:outline-none transition ease-in-out duration-100`,
                fullWidth && tw`w-full`,
                style === 'normal' && tw`p-1 background[#03c4fa] hover:bg-blue-300`
            ]}
            onClick={onClick}
        >
            <span
                css={[
                    tw`block cursor-pointer`,
                    style === 'yellow' && tw`background[#fdea36]`,
                    style === 'normal' &&
                        tw`background[#0086e7] transform-origin[50] transform[perspective(250px) rotateX(16deg)]`
                ]}
            >
                <span
                    css={[
                        tw`block cursor-pointer truncate uppercase tracking-wide leading-none pt-2 pb-1 px-6 font-semibold font-burbank`,
                        size === 'large' && tw`text-2xl`,
                        size === 'medium' && tw`text-lg`,
                        size === 'small' && tw`text-sm`,
                        style === 'yellow' && tw`color[#151a30]`,
                        style === 'normal' &&
                            tw`color[#3ae3fa] group-hover:text-white transform-origin[50] transform[perspective(245px) rotateX(0)]`
                    ]}
                >
                    {label}
                </span>
            </span>
        </button>
    );
};

export default Button;
