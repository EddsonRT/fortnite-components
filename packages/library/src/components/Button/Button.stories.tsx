import React from 'react';
import { Story, Meta } from '@storybook/react';

import Button from './Button';
import { ButtonProps } from './types';

export default {
    title: 'Components/Button',
    component: Button,
    decorators: [
        (Story) => (
            <div tw='max-width[16rem]'>
                <Story />
            </div>
        )
    ]
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

const label = 'Button';
const size = 'medium';
const fullWidth = true;

export const Small = Template.bind({});
Small.args = { label, size: 'small', style: 'lightblue' };

export const Medium = Template.bind({});
Medium.args = { label, size: 'medium', style: 'lightblue' };

export const Large = Template.bind({});
Large.args = { label, size: 'large', style: 'lightblue' };

export const LightBlue = Template.bind({});
LightBlue.args = { label, size, fullWidth, style: 'lightblue' };

export const Blue = Template.bind({});
Blue.args = { label, size, fullWidth, style: 'blue' };

export const Orange = Template.bind({});
Orange.args = { label, size, fullWidth, style: 'orange' };

export const Yellow = Template.bind({});
Yellow.args = { label, size, fullWidth, style: 'yellow' };

export const White = Template.bind({});
White.args = { label, size, fullWidth, style: 'white' };

export const Black = Template.bind({});
Black.args = { label, size, fullWidth, style: 'black' };

export const Custom = Template.bind({});
Custom.args = {
    label,
    size,
    fullWidth,
    style: 'custom',
    textColor: '#ffffff',
    bgColor: '#7e1c1c',
    borderColor: '#c74e4e'
};
