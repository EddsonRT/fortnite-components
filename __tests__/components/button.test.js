import { shallow } from 'enzyme'

import Button from '../../src/components/Button'

describe('Button Component Render', () => {
  it.skip('renders without crashing', () => {
    shallow(<Button />)
  })

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Button />)

    expect(component).toMatchSnapshot()
  })
})
