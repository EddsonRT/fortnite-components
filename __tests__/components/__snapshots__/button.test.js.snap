// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Render should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
